tlvc/arch/i86/drivers/block/ll_rw_blk.c

```diff
@@ -181,27 +173,14 @@ static void make_request(unsigned short major, int rw, struct buffer_head *bh)
 #endif
     ext_buffer_head *ebh = EBH(bh);
 
-#ifdef CONFIG_FAR_BUFHEADS
     debug_blk("BLK (%d) %lu %s %lx:%x\n", major, buffer_blocknr(bh), rw==READ? "read": "write",
-	buffer_seg(bh), buffer_data(bh));
-#else
-    debug_blk("BLK (%d) %lu %s %x:%x\n", major, buffer_blocknr(bh), rw==READ? "read": "write",
-	buffer_seg(bh), buffer_data(bh));
-#endif
+	(unsigned long)buffer_seg(bh), buffer_data(bh));
 
-#ifdef CONFIG_BLK_DEV_CHAR
-    /* remember: b_count is the # of references! */
-    //printk("Mreq: bh %04x blk %ld b_count %d\n", (unsigned int) bh, ebh->b_blocknr, ebh->b_count);
-    if (ebh->b_count & 0x80) {
-	blks = (sector_t)(ebh->b_count & 0x7f); /* # of sectors */
-	ebh->b_count = 1;
-    }
-#endif
 #ifdef BDEV_SIZE_CHK	/* NOTE: not updated for raw IO */
     sector_t count = BLOCK_SIZE / FIXED_SECTOR_SIZE;	/* FIXME must move to lower level*/
     sector_t sector = buffer_blocknr(bh) * count;
     if (blk_size[major])
-	if (blk_size[major][MINOR(buffer_dev(bh)&0x7fff)] < (sector + count) >> 1) {
+	if (blk_size[major][MINOR(buffer_dev(bh))] < (sector + count) >> 1) {
```

Comment:
Just an FYI that the `blk_size` code is a bit messed up along with BDEV_SIZE_CHK and should be considered for removal. Note that `blk_size` is declared `int` (a carryover from Linux where sizeof(int) == 32) and count and sector are sector_t (long). Any drive over 65Mb will cause integer overflow. The mechanism also requires drivers to allocate a blk_size for each partition on each minor drive, using excess storage as you previously brought up as a problem with the BIOSHD driver and its MINOR_SHIFT == 5.

Finally, there's a major bug in the genhd.c extended partition recognition code if the `size` member of struct gendisk isn't initialized (line 262). That struct member is also an `int` and thus useless for large partitions. At a bare minimum genhd.c's line 262 should also be ifdef BDEV_SIZE_CHK.

I ended up deleting all of it and saving kernel data space in ELKS (along with also changing BIOSHD MINOR_SHIFT to 3 which freed up 1K bytes kernel data, thanks to your commenting about it in the source some time ago).

Reply:
Thanks - and yes, the BDEV_SIZE_CHK stuff has never been used in TLVC and is on its way out (the `&0x7fff` was a bulk change as part of an experiment, now reverted). Which means there is no size check at this time. To be fixed.

The extended partition code has never been used (in TLVC) and I've wanted to get rid of it. Thanks for the hint - it's now ifdef'ed.

The MINOR_SHIFT reduction is on my list - thanks for the reminder.